### PR TITLE
SCB-2360 Provides a unified switch to control exception stack printing

### PR DESCRIPTION
**Pull request by develpoerX**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: handlers/handler-bizkeeper/src/main/java/org/apache/servicecomb/bizkeeper/BizkeeperCommand.java [hunk 1]
```diff
@@ -65,7 +67,12 @@ protected String getCacheKey() {
         f.onNext(FallbackPolicyManager.getFallbackResponse(type, cause, invocation));
         f.onCompleted();
       } catch (Exception e) {
-        LOG.warn("fallback failed due to:" + e.getMessage());
+        if (DynamicPropertyFactory.getInstance().getBooleanProperty(Const.PRINT_SENSITIVE_ERROR_MESSAGE,
+            false).get()) {
+          LOG.warn("fallback failed due to:" + e);
```
Comment (liubao68) [inline handlers/handler-bizkeeper/src/main/java/org/apache/servicecomb/bizkeeper/BizkeeperCommand.java:72]: LOG.warn("fallback failed due to:" ,  e);

File: handlers/handler-bizkeeper/src/main/java/org/apache/servicecomb/bizkeeper/BizkeeperCommand.java [hunk 2]
```diff
@@ -65,7 +67,12 @@ protected String getCacheKey() {
...
+        } else {
+          LOG.warn("fallback failed due to:" + e.getMessage());
```
Comment (liubao68) [inline handlers/handler-bizkeeper/src/main/java/org/apache/servicecomb/bizkeeper/BizkeeperCommand.java:74]: LOG.warn("fallback failed due to {}", e.getMessage());

File: core/src/main/java/org/apache/servicecomb/core/ConfigurationSpringInitializer.java [hunk 3]
```diff
@@ -231,15 +232,21 @@ private static void addMappingToSpring(Environment environment) {
...
+    ConfigurableEnvironment ce = (ConfigurableEnvironment) environment;
+    if (configCenterConfigurationSource == null) {
+      return;
+    }
+    try {
+      ce.getPropertySources().addFirst(new MapPropertySource("dynamic-source", dynamicData));
+    } catch (Exception e) {
+      if (DynamicPropertyFactory.getInstance().getBooleanProperty(Const.PRINT_SENSITIVE_ERROR_MESSAGE,
+          false).get()) {
+        LOGGER.warn("set up spring property source failed. msg", e);
```
Comment (liubao68) [inline core/src/main/java/org/apache/servicecomb/core/ConfigurationSpringInitializer.java:247]: LOGGER.warn("set up spring property source failed. ", e);

**Review by liubao68 — APPROVED** [review]

File: core/src/main/java/org/apache/servicecomb/core/handler/impl/ProducerOperationHandler.java [hunk 4]
```diff
@@ -125,10 +128,15 @@ public Response doInvoke(Invocation invocation, SwaggerProducerOperation produce
       invocation.onBusinessMethodFinish();
       invocation.onBusinessFinish();
     } catch (Throwable e) {
-      if (shouldPrintErrorLog(e)) {
-        invocation.getTraceIdLogger().error(LOGGER, "unexpected error operation={}, message={}",
-            invocation.getInvocationQualifiedName(),
-            org.apache.servicecomb.foundation.common.utils.ExceptionUtils.getExceptionMessageWithoutTrace(e));
+      if (DynamicPropertyFactory.getInstance().getBooleanProperty(Const.PRINT_SENSITIVE_ERROR_MESSAGE,
```
Comment (lizeze) [inline core/src/main/java/org/apache/servicecomb/core/handler/impl/ProducerOperationHandler.java:131]: 请问为什么`servicecomb.error.printSensitiveErrorMessage`==`true`的时候不使用用`invocation.getTraceIdLogger().error`打印日志呢？这样打出的异常堆栈没有`trace id`,根据`trace id` 查询错误日志的时候会漏掉这个信息呢

Comment (Shoothzj) [reply core/src/main/java/org/apache/servicecomb/core/handler/impl/ProducerOperationHandler.java:131]: @lizeze I think it's a bug, and I create #3220 to fix this

**Pull request merged**